ddtrace/tracer/rules_sampler.go

```diff
@@ -154,6 +163,29 @@ func RateRule(rate float64) SamplingRule {
 	}
 }
 
+func TagsResourceRule(tags map[string]*regexp.Regexp, resource, name, service string, rate float64, targetRoot bool) SamplingRule {
+	return SamplingRule{
+		Service:        globMatch(service),
+		Name:           globMatch(name),
+		Resource:       globMatch(resource),
+		Rate:           rate,
+		TargetSpanRoot: targetRoot,
+		Tags:           tags,
+		ruleType:       SamplingRuleTrace,
+	}
+}
+
+func SpanTagsResourceRule(tags map[string]*regexp.Regexp, resource, name, service string, rate float64) SamplingRule {
+	return SamplingRule{
+		Service:  globMatch(service),
+		Name:     globMatch(name),
+		Resource: globMatch(resource),
+		Rate:     rate,
+		Tags:     tags,
+		ruleType: SamplingRuleSpan,
+	}
+}
```

Comment:
Do these need to be public?

Reply:
Yes, because we have a tracer StartOption : `func WithSamplingRules(rules []SamplingRule)`